packages/providers/src/provider.ts

```diff
@@ -382,14 +409,31 @@ export default class Provider {
   /**
    * Create GraphQL client and set operations.
    *
-   * @param url - The URL of the Fuel node
-   * @param options - Additional options for the provider
    * @returns The operation SDK object
    */
-  private createOperations(url: string, options: ProviderOptions = {}) {
-    this.url = url;
-    const gqlClient = new GraphQLClient(url, options.fetch ? { fetch: options.fetch } : undefined);
-    return getOperationsSdk(gqlClient);
+  private createOperations() {
+    const fetchFn = Provider.getFetchFn(this.options);
+    const gqlClient = new GraphQLClient(this.url, {
+      fetch: (url: string, requestInit: FetchRequestOptions) =>
+        fetchFn(url, requestInit, this.options),
+    });
+
+    // @ts-expect-error This is due to this function being generic. Its type is specified when calling a specific operation via provider.operations.xyz.
+    return getOperationsSdk((query, vars) => {
+      const isSubscription =
+        (query.definitions.find((x) => x.kind === 'OperationDefinition') as { operation: string })
+          ?.operation === 'subscription';
+
+      return isSubscription
+        ? fuelGraphQLSubscriber({
+            url: this.url,
+            query,
+            fetchFn: (url, requestInit) =>
+              fetchFn(url as string, requestInit as FetchRequestOptions, this.options),
+            variables: vars as Record<string, unknown>,
+          })
+        : gqlClient.request(query, vars);
+    });
   }
```

Comment:
I prefer it more in the way I wrote it. I'm not really a fan of putting callbacks into variables in cases like these because you have to specify the types of the parameters to conform with `getOperationsSdk`, whereas when you define the callback directly in it, you get the type inference implicitly.

Reply:
I prefer it more as it is currently. I'm not really a fan of putting callbacks into variables in cases like these because you have to specify the types of the parameters to conform with `getOperationsSdk`, whereas when you define the callback directly in it, you get the type inference implicitly.

Reply:
@nedsalk If you don't specify, all types should be inferred just the same.

@Torres-ssf NIT'ing in this direction, I'd even go as far as formatting this:

```ts
const opDefinition = query.definitions.find((x) => x.kind === 'OperationDefinition') as { operation: string };
const isSubscription = opDefinition?.operation === 'subscription';

 if (isSubscription) {
  ...
```

I usually favor readability. We read code way more than we write it, and increased readability **_now_** usually means maintainability speed in the **_future_**. The opposite can be inversely proportional.

Reply:
@arboleya agreed.

Reply:
It seems like preference, as I find both implementations equally readable. I implemented your suggestion.